src/interactions/slashCommands/options/user.ts

```diff
@@ -0,0 +1,10 @@
+import { ApplicationCommandOptionType } from 'discord-api-types/v8';
+import { SlashCommandOptionBase } from '../mixins/CommandOptionBase';
+
+export class SlashCommandUserOption extends SlashCommandOptionBase {
+	protected override type = ApplicationCommandOptionType.USER as const;
+
+	public constructor() {
+		super(ApplicationCommandOptionType.USER);
+	}
+}
```

Comment:
I just realised you could do:

```typescript
function createSlashCommandOption<T extends ApplicationCommandOptionType>(type: T) {
	return class extends SlashCommandOptionBase {
		protected override type = type as T;

		public constructor() {
			super(type);
		}
	}
}
```

And then do:

```typescript
export const SlashCommandUserOption = createSlashCommandOption(ApplicationCommandOptionType.USER);
```

Reply:
Could I? Yes. In the event there's special options for each of the option tho, I'd rather keep them as is (instead of then doing `class extends createSlashCommandOption(T)`, although that's also valid)

CC @iCrawl thoughts?

Reply:
I think Kyras approach makes more sense here overall.